src/nvim/os/stdpaths.c

```diff
@@ -0,0 +1,88 @@
+#include "nvim/os/os.h"
+#include "nvim/strings.h"
+#include "nvim/path.h"
+#include "nvim/garray.h"
+
+typedef enum {
+  config_home,
```

Comment:
WIll do. However, I want to contain all references to the XDG spec in this module and make its public interface be something general/nvim specific..